python/pyspark/pandas/data_type_ops/num_ops.py

```diff
@@ -213,37 +214,44 @@ def abs(self, operand: IndexOpsLike) -> IndexOpsLike:
             F.abs(operand.spark.column), field=operand._internal.data_fields[0]
         )
 
+    def eq(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
+        try:
+            _sanitize_list_like(right)
+        except TypeError:
```

Comment:
Can you update your PR description, add a bit of docstring in `_is_extension_dtypes`? I can't follow why we need try-except here but not in `ne`.

Reply:
Because currently `eq` only support for list-like object properly.

Added some comments and test for `NullOps` as well.